domain/test/usecases/authentication/create_permanent_token_interactor_test.dart

```diff
@@ -0,0 +1,114 @@
+      expect(result, Right<Failure, Success>(AuthenticationViewState(permanentToken)));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithWrongUrl', () async {
+      when(authenticationRepository.createPermanentToken(wrongUrl, userName1, password1))
+          .thenThrow(ServerNotFound());
+      final result = await createPermanentTokenInteractor.execute(wrongUrl, userName1, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(ServerNotFound())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithWrongUserName', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName2, password1))
+          .thenThrow(BadCredentials());
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName2, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(BadCredentials())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithWrongPassword', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password2))
+          .thenThrow(BadCredentials());
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName1, password2);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(BadCredentials())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithConnectionError', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password1))
+          .thenThrow(ConnectError());
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName1, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(ConnectError())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithUnknownError', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password1))
+          .thenThrow(UnknownError('unknown error'));
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName1, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(UnknownError('unknown error'))));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithInternalServerError', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password1))
+          .thenThrow(InternalServerError());
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName1, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(InternalServerError())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithMissingRequiredFieldError', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, UserName(''), password1))
+          .thenThrow(MissingRequiredFields());
```

Comment:
Really, real function will throw this Error?

Reply:
<img width="603" alt="Screen Shot 2020-11-06 at 09 54 15" src="https://user-images.githubusercontent.com/16035728/98320832-4db4dc80-2016-11eb-8b95-81f031330077.png">

Yes, we have that in our api document. I also tested with postman, same response is returned when I removed `label` data in `createPermanentToken` api

Reply:
same, let me add this case in `AuthenticationDataSource`.